### PR TITLE
[RTNews] Add extractor

### PR DESCRIPTION
**Pull request by Ashish0804**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [ ] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [ ] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [ ] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Closes https://github.com/yt-dlp/yt-dlp/issues/2371

### CODE REVIEW

File: yt_dlp/extractor/rtnews.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    def _real_extract(self, url):
+        id = self._match_id(url)
+        display_id = id.split('-', maxsplit=1)[0]
+        webpage = self._download_webpage(url, display_id)
+        if 'og:video' not in webpage:
+            self.raise_no_formats('No video/audio found at the provided url.', expected=True)
+
+        return {
+            'id': id,
+            'display_id': display_id,
```
Comment (MinePlayersPE) [inline yt_dlp/extractor/rtnews.py:44]: Isn't it the opposite? https://www.rt.com/shows/in-question/535980-yt-dlp-test/ works, and the "real" slug name is only known by looking through the final redirect URL (which isn't being done here)

Comment (Ashish0804) [reply yt_dlp/extractor/rtnews.py:44]: bruh lol.
Thanks for catching this.

**Comment by pukkandan** [comment]
The BaseIEs seems unnecessary?

**Comment by Ashish0804** [comment]
Currently yes. But I'm sure I might have missed some url/alt/category somewhere and then it will be easier add support for them later on. 

That being said, I'll see if I can merge them into 1 base class.

**Comment by pukkandan** [comment]
having a baseclass for future-proofing is not a good idea imo. The parts of the code that will be shared with a related extractor will probably not be the parts you expect

**Pull request merged**